src/test/java/BoardTest.java

```diff
@@ -1,14 +1,82 @@
 import org.junit.Test;
 
-import static org.junit.Assert.assertEquals;
+import java.util.List;
+
+import static org.junit.Assert.*;
 
 public class BoardTest {
     @Test
-    public void testAppHasABoard() {
-        Board classUnderTest = new Board();
-        assertEquals("Hello world.", classUnderTest.getBoard());
+    public void testNewBoardIsEmpty() {
+        Board board = new Board();
+        assertTrue(board.isEmpty());
+    }
+
+    @Test
+    public void testAddAMarkToBoard() {
+        Board board = new Board();
+        board.markBoard(1, "X");
+        assertEquals("X", board.getMarkAtPosition(1));
+    }
+
+    @Test
+    public void testKnowThatAMoveIsValid() {
+        Board board = new Board();
+        board.markBoard(1, "X");
+        assertTrue(board.isValidMove(2));
+    }
+
+    @Test
+    public void testKnowsThatAMoveIsNotValid() {
+        Board board = new Board();
+        board.markBoard(1, "X");
+        assertFalse(board.isValidMove(1));
+    }
+
+    @Test
+    public void testKnowsThatAPlayerHasWon() {
+        Board board = new Board();
+        board.markBoard(1, "X");
+        board.markBoard(2, "X");
+        board.markBoard(3, "X");
+        assertTrue(board.playerHasWon("X"));
     }
-}
 
+    @Test
+    public void testKnowsThatThePlayerHasNotWon() {
+        Board board = new Board();
+        board.markBoard(1, "X");
+        board.markBoard(2, "X");
+        assertFalse(board.playerHasWon("X"));
+    }
 
+    @Test
+    public void testKnowsThatTheBoardIsFull() {
```

Comment:
In this test (and the rest of the tests where you add more than one mark), the board you end up with is an 'illegal' one, where only one player adds marks to the board.

Although this invalid scenario is practically allowed by your board implementation, it would be better if your tests worked through valid scenarios.
Unless you are specifically checking what happens when you try adding the same mark twice in a row,  but let's not go there.

Reply:
✅ PR #6